RecoParticleFlow/PFClusterProducer/interface/PFEcalBarrelRecHitCreator.h

```diff
@@ -0,0 +1,254 @@
+
+      iEvent.getByToken(recHitToken_,recHitHandle);
+      for(const auto& erh : *recHitHandle ) {      
+	const DetId& detid = erh.detid();
+	auto energy = erh.energy();
+	auto time = erh.time();
+        bool hi = isHighInterest(detid);
+
+	const CaloCellGeometry * thisCell= ecalGeo->getGeometry(detid);
+  
+	// find rechit geometry
+	if(!thisCell) {
+	  edm::LogError("PFEcalBarrelRecHitCreator")
+	    <<"warning detid "<<detid.rawId()
+	    <<" not found in geometry"<<std::endl;
+	  continue;
+	}
+
+	out->emplace_back(thisCell, detid.rawId(),Layer,
+			   energy); 
+
+        auto & rh = out->back();
+	
+	bool rcleaned = false;
+	bool keep=true;
+
+	//Apply Q tests
+	for( const auto& qtest : qualityTests_ ) {
+	  if (!qtest->test(rh,erh,rcleaned,hi)) {
+	    keep = false;	    
+	  }
+	}
+	  
+	if(keep) {
+	  rh.setTime(time);
+	  rh.setDepth(1);
+	} 
+        else {
+	  if (rcleaned) 
+	    cleaned->push_back(std::move(out->back()));
+          out->pop_back();
+        }
+      }
+    }
+
+    void init(const edm::EventSetup &es) {
+
+      edm::ESHandle<CaloGeometry> pG;
+      es.get<CaloGeometryRecord>().get(pG);   
+
+      //  edm::ESHandle<CaloTopology> theCaloTopology;
+      //  es.get<CaloTopologyRecord>().get(theCaloTopology);     
+
+      edm::ESHandle<EcalTrigTowerConstituentsMap> hetm;
+      es.get<IdealGeometryRecord>().get(hetm);
+      eTTmap_ = &(*hetm);
+  
+      const EcalBarrelGeometry * myEcalBarrelGeometry = dynamic_cast<const EcalBarrelGeometry*>(pG->getSubdetectorGeometry(DetId::Ecal,EcalBarrel));
+      // std::cout << "Got the geometry " << myEcalBarrelGeometry << std::endl;
+      const std::vector<DetId>& vec(myEcalBarrelGeometry->getValidDetIds(DetId::Ecal,EcalBarrel));
+      unsigned size=vec.size();    
+      for(unsigned ic=0; ic<size; ++ic) 
+        {
+          EBDetId myDetId(vec[ic]);
+          int crystalHashedIndex=myDetId.hashedIndex();
+          /// barrelRawId_[crystalHashedIndex]=vec[ic].rawId();
+          // save the Trigger tower DetIds
+          EcalTrigTowerDetId towid= eTTmap_->towerOf(EBDetId(vec[ic]));
+          int TThashedindex=towid.hashedIndex();      
+          theTTDetIds_[TThashedindex]=towid;                  
+          crystalsinTT_[TThashedindex].push_back(crystalHashedIndex);
+        }
+      unsigned nTTs=theTTDetIds_.size();
+
+      //  EBDetId myDetId(-58,203);
+      ////  std::cout << " CellID " << myDetId << std::endl;
+      //  EcalTrigTowerDetId towid= eTTmap_->towerOf(myDetId);
+      ////  std::cout << " EcalTrigTowerDetId ieta, iphi" << towid.ieta() << " , " << towid.iphi() << std::endl;
+      ////  std::cout << " Constituents of this tower " <<towid.hashedIndex() << std::endl;
+      //  const std::vector<int> & xtals(crystalsinTT_[towid.hashedIndex()]);
+      //  unsigned Size=xtals.size();
+      //  for(unsigned i=0;i<Size;++i)
+      //    {
+      //      std::cout << EBDetId(barrelRawId_[xtals[i]]) << std::endl;
+      //    }
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant